### ISSUE TITLE
Support for OS 5.0/5.1

### ISSUE THREAD

**Issue by GamingWithAlanX** [ndless-nspire/Ndless#177]
I just got a TI Nspire CXII CAS and I want to install Ndless but I can’t downgrade bc the new os has a feature exclusive to my device

**Comment by Vogtinator** [comment]
Currently not, but it's definitely planned for the future.

**Comment by GamingWithAlanX** [comment]
How long in the future will that be?

**Comment by Pien324** [comment]
I also have recently bought TI Nspire CX II CAS and was wondering if Ndless will ever be available for it. Any updates? Thanks

**Comment by Vogtinator** [comment]
Yes, we're working on it. There's always more work needed if there are significant hardware changes.

**Comment by Jukelyn** [comment]
I just got mine as well lol. The Nspire CX II CAS

**Comment by bobjohncena2** [comment]
Is there any way one could help with this? I personally have an OS 5.0 calculator, and can do some programming. However, the repository hasn't been updated in a while, leading me to think that development is going on in some other place, to be published here?

**Comment by Vogtinator** [comment]
Currently focus is on getting OS 5.0 emulation working in firebird.

**Comment by saminlatif50** [comment]
Would it be possible to sign up for an email notification or something once support for 5.0 comes out? Thanks in advance!

**Comment by lights0123** [comment]
@saminlatif50 you can hit the Watch button on this thread (it may already be enabled since you have commented), and you'll get an email when this issue is updated.

**Comment by bobjohncena2** [comment]
Wait, but isn't ndless required to get boot images for firebird? I am wondering how you are emulating it without an existing jailbreak.

**Comment by codeHusky** [comment]
Considering picking up a CX ii CAS myself, but gonna wait for you folks to support it first! Keep up the good work as usual.

**Comment by codeHusky** [comment]
@Vogtinator For a first time contributor, how would I be able to help with bringing ndless to 5.0? I imagine it's working on firebird?

**Comment by Vogtinator** [comment]
> @Vogtinator For a first time contributor, how would I be able to help with bringing ndless to 5.0?

Unfortunately there's not much you can do. If you know how to pick OS files apart you can help reverse-engineering and document your findings on https://hackspire.org.

> I imagine it's working on firebird?

Not yet - there's the issue with bootstrapping as mentioned above, which means that without Ndless you can't get firebird to boot the OS and without emulation developing Ndless is harder.

**Comment by codeHusky** [comment]
Regarding the firebird thing, I meant that _working on firebird_ would be the thing I'd need to do to help out. But yeah, I can definitely chew away at those OS files. Expect some contribs on hackspire at some point.

**Comment by codeHusky** [comment]
@Vogtinator I put in for an account on Hackspire 2 days ago. Any way to get in touch with someone to get me approved sooner?

**Comment by Vogtinator** [comment]
> @Vogtinator I put in for an account on Hackspire 2 days ago. Any way for me to get in touch with someone to get approved sooner?

@jimbauwens is admin, so maybe he can help you.

Feel free to write down any findings in the GitHub wiki until your account is working.

**Comment by jimbauwens** [comment]
Hi, your account should be now created!

Kind regards,
Jim

**Comment by jackschedel** [comment]
any ETA when this will be working on the cx 2 cas?

**Comment by debrouxl** [comment]
Across all platforms and makers, the rule for jailbreak-type frameworks is not to post information about ETA. Ndless is usually no exception to that general rule :)

**Comment by adriweb** [comment]
(title to update to "5.1" anyway :P)

**Comment by Haptikal** [comment]
When can we expect a release for OS 5.0?

**Comment by lights0123** [comment]
@Haptikal https://github.com/ndless-nspire/Ndless/issues/177#issuecomment-522353238 https://github.com/ndless-nspire/Ndless/issues/173#issuecomment-527009275

**Comment by erxk** [comment]
Is there anything new about the 5.0 OS, because it's not possible to downgrade the OS, or am I wrong there?

**Comment by GamingWithAlanX** [comment]
### 

> Is there anything new about the 5.0 OS, because it's not possible to downgrade the OS, or am I wrong there?

Depends if you have an Nspire CXII (or cas) or not. 5.0 OS brings the calculator it’s functionality.

**Comment by erxk** [comment]
I have the Ti nspire CX II-t

**Comment by GamingWithAlanX** [comment]
> I have the Ti nspire CX II-t

Ok, then you can’t downgrade because 5.0 OS let’s the calculator actually work.

**Comment by erxk** [comment]
Okay ty, but do you know sonething vout the ndless release gor my version?

**Comment by GamingWithAlanX** [comment]
We can only wait and hope Fabien is able to do it. No pressure to him.

On Sep 30, 2019, at 3:51 PM, erxk <notifications@github.com> wrote:

﻿

Okay ty, but do you know sonething vout the ndless release gor my version?

**Comment by Vogtinator** [comment]
If someone knows a bit about lowlevel usb stuff and would like to help speed this up a bit, here's how:
- Build firebird from the [cxIIusb](https://github.com/Vogtinator/firebird-fork/tree/cxIIusb) branch (you need both the GUI and the headless variant)
- Generate a flash image for the CX II using the GUI (it's not a proper CX II flash, but for this it's enough)
- Extract a OS 5.1 file, then extract the installer.img inside to get image.bin
- Run `firebird-headless --boot1 /dev/null --flash path/to/flash.img --rampayload-address 0x13200000 --rampayload path/to/image.bin`
- Change firebird in a way that packets can be received from the OS properly (no stream of `WW b00001b0 = 1000` all the time after it's booted)

I'm making progress, but as I'm fairly busy currently not at the speed I'd like.

**Comment by SirLancelot404** [comment]
Hey, this may be pointless but I was just wondering where I could go to learn whatever you need help with so that maybe at some point I could be of some help. I know little to zero about programming and this stuff but I really want to play pokemon in my math class, and am very devoted to the idea of speeding this process up. (I accidentally bought a TI-Nspire CX II thinking I could play games on it )If it's too complicated and I wouldn't be of any help thats all fine too.

**Comment by jackschedel** [comment]
Lancelot I'd guess that the type of coding for this project is very low level, meaning it'd be very difficult for even someone familiar with more standard types of coding, let alone someone with no experience at all. The main problem with trying to learn this type of stuff with no experience is that usually the documentation/tutorials are written/created assuming a knowledge of coding. Basically if u wanted to learn it, youd probably have to go and learn a completely unrelated language to this project first simply so you understand the more basic/intermediate things. This is all just an educated guess so if anyone knows of some sort of beginner friendly tutorials/docs to this stuff be sure to correct me.

**Comment by GamingWithAlanX** [comment]
Could you send me one of the tutorials? I do know how to code, just I’m unable to find one.

On Oct 2, 2019, at 8:50 AM, Jack Schedel <notifications@github.com> wrote:

﻿

Lancelot I'd guess that the type of coding for this project is very low level, meaning it'd be very difficult for even someone familiar with more standard types of coding, let alone someone with no experience at all. The main problem with trying to learn this type of stuff with no experience is that usually the documentation/tutorials are written/created assuming a knowledge of coding. Basically if u wanted to learn it, youd probably have to go and learn a completely unrelated language to this project first simply so you understand the more basic/intermediate things. This is all just an educated guess so if anyone knows of some sort of beginner friendly tutorials/docs to this stuff be sure to correct me.

**Comment by SirLancelot404** [comment]
It's no problem I was kinda expecting that as a response. But I was wondering if you think I should return my calculator while I still can or I should wait until it's compatible because I don't really know how long this would take, and I'm sure you guys don't really know either.

**Comment by Sanjit1** [comment]
Any Updates?

[![best-regards](https://img.shields.io/badge/Best%20regards-Sanjit-accff2?style=for-the-badge)](https://www.sanjit.wtf)

**Comment by erxk** [comment]
I think if there would be any relevant information about the progress they would've posted it here, or just publish the new version

**Comment by Nbobito** [comment]
I don't have a TI-Nspire cx II, but is it possible to make it boot ti nspire cx II os on it, using something like nboot?

**Comment by Vogtinator** [comment]
No, the hardware is not compatible.

**Comment by MrPoptartandMilk** [comment]
Do you have a estimated release date yet

**Comment by hacker4399** [comment]
there new os today is 4.5.2 now i wonder there possiable fix 4.5.2 ndless????

**Comment by adriweb** [comment]
No yet, see here: https://github.com/ndless-nspire/Ndless/issues/186

Good thing an Ndless version for 4.5.1 wasn't released otherwise it would very probably have gotten blocked on 4.5.2 and would have only lived a few weeks/months...

**Comment by BotApril** [comment]
Is there any updates?  I have the same problem i dont know if i should buy a new one hoping it has a compatible os or just wait

**Comment by dfghjhgfdsdfghgfdsdfghfd** [comment]
Honestly i can not wait till ndless 5.0, freal tho need status report. Is it still in progress or what xP

**Comment by dfghjhgfdsdfghgfdsdfghfd** [comment]
@Vogtinator

**Comment by codeHusky** [comment]
as mentioned previously, support for 5.0 is a work in progress. it's going to be a while since the 5.0 release (and cx II hardware in general) is significantly different than the original cx.

**Comment by Vogtinator** [comment]
> it's going to be a while since the 5.0 release (and cx II hardware in general) is significantly different than the original cx.

As Firebird is meanwhile able to boot CX II OSs, that part of the problem is solved meanwhile. Now it's the usual process of finding an exploitable bug and adding support to Ndless. There are more OS flavors to deal with this time though (4.5.2 CX, 4.5.2 CX CAS, 5.1 CX II, 5.1 CX II CAS, 5.1 CX II-T and 5.1 CX II CAS-T). That's going to take some time.

**Comment by KhiemmyT** [comment]
I think I've gone crazy with Jen's Minecraft 2D.
I've been losing my sanity ever since I upgraded to 4.5.2.
Please send help.

**Comment by Gh0stgamer92** [comment]
I hope you finish it soon, sadly i can’t help you cause I’m 14 y/o and literally don’t know anything about coding and this stuff, but I wanna play Pokémon on my calculator ;)

**Comment by Gh0stgamer92** [comment]
Do you think you’ll finish it till Christmas?

**Comment by MegaArthur** [comment]
@Gh0stgamer92 

Quoting: "The amount of time Fabian has to spend reading the endless repeated dumb Github / e-mail / forum thread questions, and replies, about Ndless 4.5.1 / 5.0 / 5.1, and replying himself to a subset of these, reduces the amount of time available for working on those things."

**Comment by Christian-Moore** [comment]
This is true. I’d ask more questions, and help, but unfortunately I don’t quite have the prerequisite knowledge. I’m working on observing and learning. Maybe it will help with my CompE courses in some manner; in which case you’ll see a lot more of me in the coming years.

**Comment by Emreal1335** [comment]
Any updates for 5.0?

**Comment by MegaArthur** [comment]
@Emreal1335

Please refer to my previous comment

**Comment by forthesafeofnealandfuller** [comment]
im guessing  it wont be too much longer now that there is a working firebird emulation right?

**Comment by vsod99** [comment]
Guys. Spamming the issue comments with requests for the release date doesn't add anything to the discussion.

It'll be done when it's done.

**Comment by YolloPlays-AI** [comment]
I just wanted to hear some news about Ndless. Is it difficult rn? What are the problems or bugs? Just a few news to keep me up-to-date, would be amazing.

Greetings

**Comment by eventhorizon02** [comment]
My skills include C/C++, assembly/machine code for MIPS, Z80 some X86 as well.
Some linux/unix. some digital electronics, micro controllers etc.
If you think I can help and can point me in the right direction I'll be happy to do so.

**Comment by Vogtinator** [comment]
> My skills include C/C++, assembly/machine code for MIPS, Z80 some X86 as well.
> Some linux/unix. some digital electronics, micro controllers etc.
> If you think I can help and can point me in the right direction I'll be happy to do so.

Experience with ARM (v5) would be useful. Getting started is quite simple: Just install OS 4.5.2 in firebird and try to get code execution by opening a custom crafted .tns file. That's a case of "draw the rest of the owl" though, the hard part is what comes after the initial setup, using tools like Ghidra to find bugs and exploiting them. If you discover something unexpected or even have a full PoC, please reach out to us in private.

**Comment by eventhorizon02** [comment]
Cool, will give it a try and will let you know if I find out anything.

**Comment by Pi737** [comment]
Can I downgrade my os to a less advanced version and get ndless with that old os? is there any way to downgrade the system os? (I'm using a CX II CAS with os 5.0.0.1683)

**Comment by YolloPlays-AI** [comment]
No. Not possible, because the 5.0 OS is the OS that makes this calc working.

**Comment by Pi737** [comment]
Thanks

**Comment by SrAnnoying** [comment]
Hey, I know I'm being annoying (sorry about that), but it's been a few months of silence, and I was wondering if there has been _any_ progress on it (i don't want to know the percentage, I just want to know if it's being worked on).

Also, is there any Discord server for ndless?

**Comment by Sanjit1** [comment]
@SrAnnoying yea I have also been wondering the same. I don't think there is a discord server and feel like there should be one.

**Comment by lights0123** [comment]
There is a subreddit: https://reddit.com/r/nspire

It's more for general TI-Nspire stuff, but there's the occasional post related to ndless, although the last non-help posts was one 2 months ago and me 8 months ago...

**Comment by badyherz** [comment]
Now I have a nspire cx cas running 4.4.0, but its the one from school. I want to buy one but probably the "new" cx ii tcas. When will there will be a ndless version for it? I want to buy it in the end of 2020.
Because I really just use it to play gbc games.

**Comment by Vogtinator** [comment]
> Hey, I know I'm being annoying (sorry about that), but it's been a few months of silence, and I was wondering if there has been any progress on it (i don't want to know the percentage, I just want to know if it's being worked on).

Yes.

> I don't think there is a discord server and feel like there should be one.

There's no discord server - it's on plain old IRC instead: #ndless-dev on freenode.

**Comment by adriweb** [comment]
(on EFNet rather than Freenode :P)

**Comment by Vogtinator** [comment]
> (on EFNet rather than Freenode :P)

Indeed - I should've waited until waking up fully before replying...

**Comment by debrouxl** [comment]
What's that frenzy with Discord ? IRC's more lightweight, and functionally just fine :)

**Comment by KhiemmyT** [comment]
Eh, it's just easier to use and more oriented for gamers but it can be used for general stuff. It still would be cool if we have one. I'll probably set one up for this topic just for fun.

**Comment by Atrate** [comment]
When it comes to tech-oriented groups I'd stay away from Discord. If you want an alternative, use Riot (Matrix client), it can be bridged with IRC. This is unrelated to this issue, if you want, you can create a Riot/Matrix community and notify us all in a new issue. I can help bridging with IRC if that were desired.

**Comment by 322997** [comment]
I would stay away from discord not only because of the egregious freedom issues with it, but also because their ToS/Guidelines are tyrranical. Chances are ndless would actually run afoul of them, because it can be considered "hacking or cracking" by some, which is explicitly prohibited by discord.

**Comment by Peteris-Zvejnieks** [comment]
If a device is needed for testing and the testing has a low probability of bricking it , I`ve got TI-nspire CX II CAS currently running 5.0.0.1683.

**Comment by 179West** [comment]
Is there any progress on ndless at all ATM? I'm thinking of upgrading to a CX II CAS from my CX CAS, and I want to keep ndless on there.

**Comment by Mikeologist** [comment]
Did you read the messages of the past few days?

**Comment by JMK19** [comment]
Hey! Sorry if this is out of the blue. Just bought the nspire CX II, and wondering if I can do anything on it, ndless or not. Does anyone know if there's anything I can do?

**Comment by lights0123** [comment]
@JMK19 assuming you mean what you can run on it now (versus "do" meaning help), you can run any Lua or TI-Basic app.

**Comment by adriweb** [comment]
And Python soon (next OS update)

**Comment by Peteris-Zvejnieks** [comment]
@JMK19  You can run some basic games from [here](https://www.ticalc.org/pub/nspire/lua/games/)

**Comment by JMK19** [comment]
Hi guys! thanks for all the tips and games, I downloaded a lot and it's really cool. I code in Python so i'm excited for Python if it's coming. Is there any progress on ndless for this calculator? Really interested in running apps like mini vMac on it.

**Comment by Kyss007** [comment]
hey guys ^^
are there any news regarding support for 5.0 and 5.1?

**Comment by KhiemmyT** [comment]
Unfortunately not

**Comment by adriweb** [comment]
At this point, there is no need for 5.0 support, and who knows, maybe even 5.1 support considering 5.2 is coming "soon" (for back-to-school anyway) and will be a major upgrade providing Python programming. There is little reason to get an Ndless release now considering nobody is at school anymore (so, way less users) and TI could have all it's time until the 5.2 release to block it by fixing whatever flaw it would be using. And so having to choose between Ndless and a new major feature would kind of suck... Anyway, that's my opinion, but I'm sure I'm not alone with it.

**Comment by SrAnnoying** [comment]
Is there any way for someone to put the CAS system on the N-spire II? I know you could do that with the first model via Ndless. That's my primary reason for waiting on Ndless.

**Comment by adriweb** [comment]
No, and this is unrelated to Ndless anyway, so...

**Comment by Vogtinator** [comment]
You can simply buy a CX II CAS. It is not the goal of Ndless to allow CAS on non-CAS hardware.

**Comment by StevenC21** [comment]
Hey guys, I'm not here to ask for a release date (I am more than aware that these things can't be rushed), but I _did_ want to say that I personally appreciate your continued development of ndless. I have 3 TI-nspire calculators (2 CX CAS - one of them is unfortunately CR 4 hardware - and 1 CX II CAS). I currently run ndless on the first two and am looking forward to running it on the third.

So, thanks guys! You've done some excellent work.

**Comment by Vogtinator** [comment]
I wonder about the "unfortunately CR 4". You should give https://github.com/ndless-nspire/Ndless/pull/202 a try.

**Comment by ION-Prod** [comment]
Y U NO HAV PATREON?!?! Like seriously, it might be worth the effort and also amazing work (even though i can't use it because i got the CX II WITHOUT KNOWING (that i couldn't get ndless on it) P.S. If you have patreon and I didn't find it I'm dumb and sorry

**Comment by AntoineQ1** [comment]
@adriweb Yeah, I think ndless with 5.2 and Python would be much better. I just wonder if Ndless will be compatible with my calculator before the end of my studies... 

So, do you think I can update it to 5.1 (and then 5.2) or it would reduces my chances of getting Ndless ? Because the work done for 5.0 may not be compatible with 5.1/5.2...

_PS : You're on TiPlanet, aren't you ? :D_

Thanks,

**Comment by adriweb** [comment]
I doubt there is any issue updating to 5.1, and later 5.2 although there will probably be an official announcement/post about it when ready :)

**Comment by officereso** [comment]
This has been in the works for a long time. Can we get some sorta update? Ndless 5.x isn’t abandoned is it?

**Comment by codeHusky** [comment]
@officereso if you took a moment and scrolled up, you'd see that a contributor to the project had posted in this thread less than a week ago. It's very much in development.

**Comment by Atrate** [comment]
Locking this issue for contribs only might be a good idea, with a message to describe why it was locked and that progress is still being made

**Comment by Vogtinator** [comment]
No, it's not abandoned.

> Locking this issue for contribs only might be a good idea, with a message to describe why it was locked and that progress is still being made

I don't really like locking conversations, it just seems wrong to me for an open project. It would probably only lead to new issues about this getting created anyway.

**Comment by KhiemmyT** [comment]
I've taken the liberty of making an (unofficial) discord server for this repo.
Here's the link: https://discord.gg/xn9jZ9p
Includes text and voice channels. I've also added roles for devs, members etc. so they'll get permissions to specific channels. However, since this is an open project, there won't be **much** difference. This is still a work in progress btw.

Q&A:
**Won't discord flag this server because it violates their TOS? (hacking/cracking):**
I highly doubt it since discord won't care about exploits on a calculator. Unless you're doing some shady stuff (like cracking games or trying to find a hack into a secure server), then you're fine.

**Isn't discord for gamers?**
While it was originally directed to the gaming community, it has grown so much that there is a wide variety of servers for pretty much anything. Teachers and students are even using it for remote learning!

**Untitled, you IDIOT. We didn't ask for a Discord server and no one wants to join it!**
I'm sure some people will prefer Discord over IRC due to its more user-friendly UI. Also, can you do instant live voice chats and have colour coded users through IRC? Probably not.

**Ok, whatever. But I'm still skeptical about this so-called Discord.**
That's completely fine. You don't have to join it. A few months ago, I thought discord was just some useless messaging application. But when someone encouraged me to use it, boy was I completely wrong. I've never stopped using it ever since and I hope you'll have the same experience as I did.

**Comment by Vogtinator** [comment]
Apparently the discussion here is going just fine, so why is there a need for discord? This seems OT.

**Comment by codeHusky** [comment]
seems kind of pointless to have anything but a ndless community discord. Plus, advertising it in this issue is probably the least effective place to do so.

**Comment by KhiemmyT** [comment]
I have way too much time on my hands. Now that you mention it, it kind of seems surreal to even make one. I'll probably just scrap the whole idea.

**Comment by anoopsk** [comment]
Hey vogtinator I found a bug in the calculator.

**Comment by Pi737** [comment]
> Hey vogtinator I found a bug in the calculator.

What?

**Comment by Vogtinator** [comment]
@anoopsk Great! There is a contact address on ndless.me where you can share details.

**Comment by anoopsk** [comment]
On the ndless.me contact page I found the email ndless.team@gmail.com I sent an email to it weeks ago but you didn't respond to it.

**Comment by anoopsk** [comment]
The title of the email was 'are you vogtinator'.

**Comment by Vogtinator** [comment]
> The title of the email was 'are you vogtinator'.

There were two mails with that subject. I responded to the first one.

**Comment by anoopsk** [comment]
Oh sorry, my mistake. I deleted the first email. Respond on the second email.

**Comment by anoopsk** [comment]
I replied to your response with info about the bug hours ago.

**Comment by Vogtinator** [comment]
> I replied to your response with info about the bug hours ago.

Yep, I got the mail but didn't have time to respond yet.

**Comment by mgitre** [comment]
Should I have my hopes up?

**Comment by Neglokit** [comment]
I am also interested, this thread has been open for more than a year. What is the planned release date?

**Comment by forthesafeofnealandfuller** [comment]
No but there is an irc I believe

**Comment by Neglokit** [comment]
What is irc?

**Comment by forthesafeofnealandfuller** [comment]
Edit: no irc but there are two forums that ndless promotes
https://tiplanet.org/forum/portal.php
https://codewalr.us/?sslRedirect
Here you can talk about ndless and such

**Comment by adriweb** [comment]
The IRC channel is #ndless-dev on EFNet servers.
But it's more related to native nspire development in general, not the specifics of ndless development itself, which isn't public until the source is put on Github here.

**Comment by forthesafeofnealandfuller** [comment]
Thanks Aiden,/!

**Comment by Neglokit** [comment]
Is the program even half done, or is it almost finished and is in it's last testing phase?

**Comment by ffs1337** [comment]
This is taking sooooo long

**Comment by RedshiftVelocities** [comment]
People! Stop being obnoxious if you want it to _ever_ get done.

**Comment by Knucklesfan** [comment]
Have you guys seriously ever seen a development process? Things take a while. Finding a bug is a  challenge in the process, not to mention implementing this bug as an exploit. This is a massive undertaking for the Ndless team, implementing support for a new calculator type is a challenge, even despite the minor changes between the cxII and the cx. We finally have some news about a potential bug (which was bound to happen to begin with) and you guys think it's gonna be an instant process. "wow, a bug was found?  that must mean porting it is a one day job!" It's not that simple.

**Comment by adriweb** [comment]
Guys.... It is much simpler than what you may all think, even though what @Knucklesfan holds true regardless.

I'll let you ponder over this:

- It is publicly known that arbitrary code execution on the CX II (so yes, exploiting something) [has been achieved] a long time ago already, since [emulation is possible](https://github.com/nspire-emus/firebird/pull/165), since it requires dumping
- In turn, emulation being possible also means that the hardware has also been [understood sufficiently well](https://hackspire.org/index.php?title=Special:Contributions/Vogtinator)
- COVID lockdown, and thus students being out-of-school, has been there for months now
- It is also [publicly known](https://education.ti.com/en/products/calculators/graphing-calculators/ti-nspire-cx-ii-cx-ii-cas/programming-in-python) that TI will release its new OS 5.2 with new features such as Python programming support

Get it? See where this is going? 

Still no? Ask yourself the question whether you really expect Ndless 5.1 to happen and inevitably be blocked within a few months as soon as 5.2 comes out, or rather expect Ndless 5.2 to happen, a bit after OS 5.2 is released. Is it obvious enough now?

**Comment by Neglokit** [comment]
So how long would it take?

**Comment by lights0123** [comment]
> a bit after OS 5.2 is released

---

@Neglokit Probably not long. Likely (hopefully) the exploit will not be discovered by TI before they release 5.2 (hence why nothing has been made public so far), so it'll just be a simple matter of discovering syscalls for the new OS, which isn't a long process.

**Comment by Peteris-Zvejnieks** [comment]
> 
> 
> Guys.... It is much simpler than what you may all think, even though what @Knucklesfan said holds true regardless.
> 
> I'll let you ponder over this:
> 
>     * It is publicly known that arbitrary code execution on the CX II (so yes, exploiting something) has been achieved a long time ago already, since [emulation is possible](https://github.com/nspire-emus/firebird/pull/165) and that requires dumping (through native code)
> 
>     * In turn, emulation being possible also means that the hardware has also been [understood sufficiently well](https://hackspire.org/index.php?title=Special:Contributions/Vogtinator)
> 
>     * COVID lockdown, and thus students being out-of-school, has been there for months now
> 
>     * It is also [publicly known](https://education.ti.com/en/products/calculators/graphing-calculators/ti-nspire-cx-ii-cx-ii-cas/programming-in-python) that TI will release its new OS 5.2 with new features such as Python programming support
> 
> 
> Get it? See where this is going?
> 
> Still no? Ask yourself the question whether you really expect Ndless 5.1 to happen and inevitably be blocked within a few months as soon as 5.2 comes out, or rather expect Ndless 5.2 to happen, a bit after OS 5.2 is released. Is it obvious enough now?

 To prevent TI from ruining the fun for us the exploit/s of the current version will be kept a secret till 5.2 comes out? Smart! But to be honest, it wasn't obvious.

**Comment by Knucklesfan** [comment]
If you've been a member of a hacking scene before (i.e 3ds or switch or something) it's actually very common to delay exploit releases until the very last minute, to ensure complete compatibility. One such example would be with the delay of ninjhax to ensure new 3ds compatibility. This is very normal.

**Comment by xkingonionx** [comment]
So I assume that the support for cx ii is largely a waiting game for 5.2 now?(other than bug fixes)

**Comment by xkingonionx** [comment]
Should I keep my os up to date until 5.2? I’m really new to this. Thanks👍

**Comment by Knucklesfan** [comment]
> Should I keep my os up to date until 5.2? I’m really new to this. Thanks👍

As is tradition, this is your own choice, but the general advice is do not update under any circumstance, the lower the firmware the more likely the exploit will function on your firmware.

**Comment by adriweb** [comment]
> the general advice is do not update under any circumstance, the lower the firmware the more likely the exploit will function on your firmware.


Until an exploit targeting this version is released anyway.

**Comment by xkingonionx** [comment]
Cool! I can’t wait!

**Comment by xkingonionx** [comment]
If I’m correct, there is a version of 5.x Ndless that is running somewhere. Has there been any benchmarking of the cx ii cas t? And if so, how does it compare to previous calcs?

**Comment by Mordy24** [comment]
when will 4.5.2.8 be out for ti nspire cx cas?

**Comment by lights0123** [comment]
@Mordy24 Dude, there's no need to spam _3_ issues, two of which are unrelated. Scroll up literally _7_ comments and see why a release is being delayed: https://github.com/ndless-nspire/Ndless/issues/177#issuecomment-649967623

**Comment by Mordy24** [comment]
ohh sorry

**Comment by xkingonionx** [comment]
as i understand it, 4.5.2.8 and all updates under 5 are for the ti nspire cx and cx cas. 5.0, 5.1, and 5.2 are for the ti nspire cx ii cas and the cx ii.

**Comment by SmileytheDuck** [comment]
I just got a TI Nspire CX II. Does anyone know any emulators that work without Ndless (im new to this).

**Comment by adriweb** [comment]
Firebird (the (only) emulator) doesn't itself require Ndless, it needs a ROM dump to work though. And dumping the ROM (boot etc.) is what requires Ndless.

**Comment by KhiemmyT** [comment]
I think he means game emulators for the ti nspire. Currently, there are no emulators that don't require ndless (they would have to run on Lua). However in September, there should be a new release of ndless for the new ti nspires.

**Comment by SmileytheDuck** [comment]
@KhiemmyT Thanks for the answer! I am looking forward for the Ndless release in September. Is there anyway to be notified of its release when it releases?

**Comment by lights0123** [comment]
@SmileytheDuck by commenting on this issue, you've automatically subscribed to all future comments on this thread. If you'd like to unsubscribe (to avoid getting a bunch of emails in a few days when the next person decides they need to ask when Ndless will be out for the CX II), you can click "Watch" at the top right of this page and select "Releases only".

**Comment by SmileytheDuck** [comment]
@lights0123 thanks

**Comment by Mordy24** [comment]
so ndless 5.0 is coming in september

**Comment by xkingonionx** [comment]
in short, most likely

**Comment by Mordy24** [comment]
is there any specific day in September it will be out

**Comment by xkingonionx** [comment]
shortly after 5.2 comes out from ti. so no. just wait for 5.2.

**Comment by adriweb** [comment]
it's not like it's known how "shortly" it will be after 5.2, but yes that's at least after its release which TI said is for September

**Comment by Neglokit** [comment]
when will it 5.0 come on it of cas?

**Comment by rem137** [comment]
It’s for when ? September?

**Comment by MininuxDev** [comment]
They already said and repeated that it's probable, but nothing is confirmed yet

**Comment by mgitre** [comment]
For God's sake people, please take 30 seconds to read the last few posts in this thread before constantly filling it with questions that have already been answered like 10 times.

**Comment by RyDawgE** [comment]
Ok, ive read most if not all of the thread, and im a bit confused. 

From what I gathered 5.0/.1/.2 May be releasing this September.

But I see others saying there's no need for it? Something about a python implementation?

So I am on 5.0.0 on my TI Nspire cx ii (NOT CAS) Do i update? Can I downgrade? Whats this python thing and why does it help? Is 5.0 even still happening?Do i stay on 5.0? I'm so lost, and I deeply apologize if I asked a question that's already been answered .

Thanks again!

**Comment by adriweb** [comment]
OS 5.2, which come out in September, will have Python programming built-in. There's no point for an Ndless version to target earlier OSes.

**Comment by Drahnoel1** [comment]
will this work for ti nspire CX II-t? Sorry if this was asked already but I didn't see it in this thread...

**Comment by RyDawgE** [comment]
> OS 5.2, which will come out in September, will have Python programming built-in. There's no point for an Ndless version to target earlier OSes.

So does that mean an emulator (gba for example)may also come out in September? Basically from what i gathered ndless is getting cucked.

**Comment by adriweb** [comment]
> will this work for ti nspire CX II-t? Sorry if this was asked already but I didn't see it in this thread...

Not sure if you're talking about Python availability (answer: yes) or Ndless availability (answer: probably I guess, at least it has always been available for all models)

> So does that mean an emulator (gba for example)may also come out in September? Basically from what i gathered ndless is getting cucked.

Well the order of things would be : OS, then Ndless, then programs. Most programs don't need to be rebuilt/adapted when a new version comes out since Ndless takes care of the heavy-lifting behind the scenes, but I don't know this time, considering the HW changed etc - we'll see?

**Comment by AntoineQ1** [comment]
**Hello,**

[To finally answer the release-date question]. I asked the Texas Instrument team Twice through different channels. 
The precise release date is unknown. It will be necessarily available before October but it seems it will be around the 15th of September.

_In fact, a pre-version has already been published and distributed to some organizations. It is therefore only a matter of time._

**Goodbye,**

**Comment by Neglokit** [comment]
> **Hello,**
> 
> [To finally answer the release-date question]. I asked the Texas Instrument team Twice through different channels.
> The precise release date is unknown. It will be necessarily available before October but it seems it will be around the 15th of September.
> 
> _In fact, a pre-version has already been published and distributed to some organizations. It is therefore only a matter of time._
> 
> **Goodbye,**

what????? release date for ndless or ti 5.2???

**Comment by AntoineQ1** [comment]
> what????? release date for ndless or ti 5.2???

The Ti-Nspire 5.2 Update

**Comment by DrD00f** [comment]
so i saw a reddit comment saying he had ndless 5.0 on a ti nspire cx II cas is this true?

**Comment by lights0123** [comment]
@DrD00f please take the time to read the whole issue. All of these questions [have already been answered](https://github.com/ndless-nspire/Ndless/issues/177#issuecomment-646142216).

**Comment by DrD00f** [comment]
yeah... I still don't get it but ill shut

**Comment by ShadyCloud117** [comment]
So I’ve read the last around 30 or so comments and I’m wondering is 5.2 os a good or a bad thing for emulation?

**Comment by Knucklesfan** [comment]
Look guys, I'll make it short and sweet. Ndless currently runs on the nspire 2, for certain. The entire reason why it isn't public is to ensure compatibility with 5.2. This has nothing to with emulation or anything, as for when the update comes out, most applications will need to be updated. Nobody publically has the ti nspire 2 running ndless on 5.0, neither on reddit or here. Internally, sure someone working on the project does but it isn't public, and it won't be public until 5.2 comes out. Please stop asking the same questions over and over again despite it being answered MULTIPLE times on this very post itself. Just wait, and you'll eventually get to use it.

**Comment by ShadyCloud117** [comment]
I was just asking if updating the os was good cause on my ti 84 plus ce got nuked by 5.6 😓

**Comment by tatey12** [comment]
When It Comes Out You Will Do Nintendo 64 Emulator On Calculator

**Comment by jkcoxson** [comment]
No, it's not powerful enough.

**Comment by logantischler0** [comment]
What about a Nintendo DS emulator? Will that ever be a thing?

**Comment by adriweb** [comment]
Here are the list of available emulators for the CX: https://tiplanet.org/forum/viewtopic.php?f=20&t=17762
We'll see if the CX II would allow for more powerful consoles... but the RAM (and ROM) capacity hasn't increased, btw...

Now, let's stop talking about off-topic things...

**Comment by RyDawgE** [comment]
> What about a Nintendo DS emulator? Will that ever be a thing?

Not only is this very off topic, but I believe you are over estimating the power of a calculator. This thing isn't a gaming PC. The best I bet this thing can run is GBA

**Comment by JackyZhou05** [comment]
Hello. Do you guys know any CAS apps for nspire cx ii since ndless doesn't function on em yet.

**Comment by ZongUin** [comment]
OS 5.2 is out

**Comment by adriweb** [comment]
Yes, don't worry, everybody who needs to know is aware :P
For those wanting a complete Python-related review, @critor published an article here with comparisons and benchmarks: https://tiplanet.org/forum/viewtopic.php?f=43&t=24224 :)

**Comment by MininuxDev** [comment]
The big question now is will ndless have to be updated for 5.2 or will there be one version for all 5.x os...

**Comment by adriweb** [comment]
Well the big answer would be simply that only 5.2 will be supported, there's literally no point in supporting the previous ones :P

**Comment by medwatt** [comment]
I have an Nspire CX CAS (not the II). I haven't installed ndless on it. I noticed on the tiplanet review that was linked 3 posts before mine that the Nspire with Ndless + Khicas (with micropython) seems much more capable even compared to what TI released for the Nspire II. Is this really the case or am I reading it wrong ?

**Comment by logantischler0** [comment]
Since TI OS 5.2 is now out, is there an approximate time frame of when Ndless 5.2 is going to be released?

**Comment by Jonson26** [comment]
"Soon"

**Comment by logantischler0** [comment]
How soon?

**Comment by NeRdTheNed** [comment]
> How soon?

https://www.youtube.com/watch?v=fq3abPnEEGE

**Comment by KhiemmyT** [comment]
So let's just say hypothetically, that ndless 5.2 was going to be released, then would it be a good idea to upgrade my OS to 5.2 now?

**Comment by Cyastis** [comment]
Hypothetically, it wouldn’t be a bad idea to download but not install 5.2.0.771 for now just in case (this way you have more options in case something goes wrong in development)

**Comment by Knucklesfan** [comment]
I would still stand and say this is a bad idea. For all we know, the Nspire development team found the bug utilized by ndless and patched it. This would leave you on a bricked calculator. (well, bricked from running ndless that is) I'd say never update until you are guaranteed there's an exploit and it still works. We have no guarantee.

**Comment by Tbonebaby** [comment]
any more updates??

**Comment by sudoBash418** [comment]
> any more updates??

If there were, you'd almost certainly see them here.
Now can y'all please stop spamming this thread with pointless questions which either have blindingly obvious answers or have literally been answered already in *this very thread?*

**Comment by adriweb** [comment]
Reminder that even if the development of Ndless itself is in theory finished, lots of testing has to take place, and for the CX II, times 3 (because there are now 3 different OSes: CX II, CX II-T, CX II CAS/II-T CAS/II-C CAS).

**Comment by Tbonebaby** [comment]
ok, sorry that i was impatient im just excited for it.

**Comment by BaiTian6641** [comment]
Want to mention that, I tear apart an CX II(Actually is "fall" apart). There have a Micron NOR Flash(I guess), and it is very common used NOR Flash. So I can try to desoldering this and put on programmer to read. Also have another 8 SOP(should be) between the NOR FLASH. And RAM chip is also separated(seem like BGA-60 or FBGA-54).

**Comment by Peteris-Zvejnieks** [comment]
> 
> 
> is there any specific release date/time for ndless

Try reading at least 3 most recent comments before commenting. There'll always be a question like yours and a reply like this.

**Comment by medwatt** [comment]
> is there any specific release date/time for ndless

At this point, I don't really care whether they release it or not. I'm bored with the waiting game and having someone with a holier than thou attitude always try to lecture people like you on why since this is something you're not paying for, you can't demand to know when there might be an update so that you can have some bit of closure.

**Comment by NeRdTheNed** [comment]
> > is there any specific release date/time for ndless
> 
> At this point, I don't really care whether they release it or not. I'm bored with the waiting game and having someone with a holier than thou attitude always try to lecture people like you on why since this is something you're not paying for, you can't demand to know when there might be an update so that you can have some bit of closure.

Dude, calm down, it'll be out when it'll be out. Someone is working on this, in their free time, so that we all get to use it. There's no need to be hostile to them.

**Comment by adriweb** [comment]
Patience, people... it's about to be here. We're still testing.

Here are some things to make you wait. Notice how it's much faster on a CX II...

![](https://thumbs.gfycat.com/PhonyRedLemur-size_restricted.gif)
![](https://thumbs.gfycat.com/GloriousSoggyDotterel-size_restricted.gif)
![](https://thumbs.gfycat.com/UniqueGrimyHare-size_restricted.gif)
![](https://thumbs.gfycat.com/LazyEmotionalAtlanticbluetang-size_restricted.gif)

**Comment by anoopsk** [comment]
I'm so excited for the release!

**Comment by anoopsk** [comment]
This is epic!

**Comment by anoopsk** [comment]
Can't wait to play GBA games at full speed with no frameskip.

**Comment by anoopsk** [comment]
The snes emulator should also be updated to work on the CX II

**Comment by Atrate** [comment]
Please do not post many comments at a time — you can edit your comments to add new text.

**Comment by jkcoxson** [comment]
Ndless was just released for the new versions!

**Comment by Vogtinator** [comment]
Yep, it's out: https://ndlessly.wordpress.com/2020/09/20/ndless-for-os-4-5-3-and-5-2-0-released-now-with-cx-ii-support/

**Comment by CerebralDatabank** [comment]
The wait is over :D

**Comment by Wibbl** [comment]
Hey,

Kleines Problem immer wenn ich es Installieren will auch meinem CX II-T kommt das

was muss ich tun?
![WhatsApp Image 2020-10-29 at 17 09 18](https://user-images.githubusercontent.com/73658755/97600714-a3cbc380-1a09-11eb-824a-ace3a2158764.jpeg)

**Comment by Drahnoel1** [comment]
Wenn das Fenster nicht weg geht versuche mal zu resetten.
Außerdem solltest du hier auf Englisch schreiben...

**Comment by Wibbl** [comment]
Unfortunately it is still as before
ups sorry

**Comment by Vogtinator** [comment]
Make sure it's actually running OS 5.2.0. If not, the installer will just not do anything.

If you've got anything new to report, please do it in a new issue.

**Comment by xkingonionx** [comment]
try stuff from this page. it could work but ive never had the issue myself
https://www.dummies.com/education/graphing-calculators/access-the-ti-nspire-maintenance-menu-to-resolve-problems/

**Comment by ArcticLemming7182613** [comment]
So, I recently bought a TI-nspire CX II CAS, I'm wondering if (since I've gotten vague answers) Ndless is supported on that specific model, or if there's a better calc to get.
(I'm mostly wondering about CX II vs CX II CAS)

**Comment by lights0123** [comment]
@ArcticLemming7182613 it doesn't matter. Ndless has always supported both CAS and non-CAS versions.

**Comment by RuralAnemone** [comment]
> @ArcticLemming7182613 it doesn't matter. Ndless has always supported both CAS and non-CAS versions.

Thank you for the quick response!


Good, because I was kinda worried when I actually looked into this if I wasted money 😋

**Comment by CircuitRCAY** [comment]
It doesn't seem like there is a Polydumper version for the Ndless 5 update...

Any other way to dump boot1/2?

**Comment by adriweb** [comment]
No, nothing yet.

**Comment by CircuitRCAY** [comment]
It looks like Polydumper supports the CXII now.
![image](https://user-images.githubusercontent.com/21284083/101705835-eccc6880-3adb-11eb-85f6-7a6a4e23209b.png)

**Comment by adriweb** [comment]
Yes, but it's not yet able to dump a part required for emulation.

**Comment by simplyrohan** [comment]
I want to bump this because I'm on version 5.1, and now that OS v6.3 has been released, I can't even update and use Ndless 6.2

**Comment by simplyrohan** [comment]
Thank you for the link! I've been looking for just that (a place to download past version).
